### PR TITLE
Da fraud proofs

### PR DESCRIPTION
**Pull request by anselld**
Hi Kenny. Your report looks great. Please review my formatting changes and let me know if you'd like any changes before I start with the editing.

### CODE REVIEW

**Comment by anselld** [comment]
Thanks, Kenny. May I start with the editing?

**Comment by hansieodendaal** [comment]
I suggest you carry on @anselld as Kenny is happy with the changes up to now.

File: src/cryptography/fraud-proofs-1/MainReport.md [hunk 1]
```diff
@@ -55,72 +54,71 @@ In the original Bitcoin whitepaper, Satoshi recognized this and introduced the c
...
-For this case, the fraud proof consists of:
-* the block itself
-
+In this case, the fraud proof consists of the block itself.
 ### Invalid Transaction if Input does not Exist
-For this case, the fraud proof consists of:
-* the entire blockchain
+In this case, the fraud proof consists of the entire blockchain.
+
+## Fraud Proofs Requiring Changes to Bitcoin Protocol
```
Comment (anselld) [inline src/cryptography/fraud-proofs-1/MainReport.md]: In the editing process, I inserted this heading to structure the following information.

File: src/cryptography/fraud-proofs-1/MainReport.md [hunk 2]
```diff
@@ -55,72 +54,71 @@ In the original Bitcoin whitepaper, Satoshi recognized this and introduced the c
...
-or 
+* a block header with transaction data relative to the client, including other transactions required to compute the Merkle root; or 
 * just a block header with no transactions.
 
 ## What are Fraud Proofs?
 
-Fraud proofs are a way to improve the security of SPV clients [[5]] by providing a mechanism for full nodes to prove that a chain is invalid irrespective of the amount of proof of work it has [[5]]. Fraud proofs could also help with the Bitcoin scaling debate as SPV clients are easier to run and could thus help with Bitcoin scalability issues. ([[6]],[[18]])
+Fraud proofs are a way to improve the security of SPV clients [[5]] by providing a mechanism for full nodes to prove that a chain is invalid, irrespective of the amount of proof of work it has [[5]]. Fraud proofs could also help with the bitcoin scaling debate, as SPV clients are easier to run and could thus help with bitcoin scalability issues. ([[6]],[[18]])
+
+## Fraud Proofs Possible within Existing Bitcoin Protocol
```
Comment (anselld) [inline src/cryptography/fraud-proofs-1/MainReport.md]: I inserted this heading to structure the information - is it okay?

File: src/cryptography/fraud-proofs-1/MainReport.md [hunk 3]
```diff
@@ -129,105 +127,105 @@ Their proposition is to generalize the entire blockchain as a state transition s
...
+Fraud proofs can be complex [[6]] and hard to implement. However, they appear to be necessary for scalability of blockchains and the security and privacy for SPV clients, since not everyone can or should want to run a full node to participate in the network. The current SPV implementations are working on improving the security and privacy of these SPV clients. Furthermore, for current blockchains, a hard or soft fork would need to be done in order to accommodate the data in the block headers.
 
-Based on [3 different fraud proof proposals](#suggested-fraud-proof-improvements) that suggest some sort of incentive for nodes that issue alert/fraud proofs, it seems likely that some sort of fraud proof providers and consumers market place will have to emerge.
+Based on [three different fraud proof proposals](#suggested-fraud-proof-improvements) that suggest some sort of incentive for nodes that issue alert/fraud proofs, it seems likely that some sort of fraud proof provider and consumer marketplace will have to emerge.
```
Comment (anselld) [inline src/cryptography/fraud-proofs-1/MainReport.md]: The link to 'three different fraud proof proposals' is unclear to me. It seems to point to this report, but I'm not sure where the three proposals are. Please help.

File: src/cryptography/fraud-proofs-1/MainReport.md (hunk 2 above)
Comment (ksloven) [reply src/cryptography/fraud-proofs-1/MainReport.md]: that is ok

**Comment by anselld** [comment]
Hi Kenny
Please could you review my editing changes and let me know if you're happy with them.
Please help with my question on line 226 regarding the link.
Thanks and regards
Denise

File: src/cryptography/fraud-proofs-1/MainReport.md (hunk 3 above)
Comment (ksloven) [reply src/cryptography/fraud-proofs-1/MainReport.md]: You are correct. I meant to mention the different proposals (universal fraud proofs, merklix trees and payment channels), although on second reading I see that only the payment channels proposal has some sort of incentive.
So it should read something like: "Based on the payment channels fraud proof proposal that suggests....."

File: src/cryptography/fraud-proofs-1/MainReport.md (hunk 2 above)
Comment (anselld) [reply src/cryptography/fraud-proofs-1/MainReport.md]: Thanks.

File: src/cryptography/fraud-proofs-1/MainReport.md (hunk 3 above)
Comment (anselld) [reply src/cryptography/fraud-proofs-1/MainReport.md]: Thanks for your help, Kenny. I've made the change, and will commit and push the report.
Regards
Denise

**Comment by anselld** [comment]
@ksloven 
Hi Kenny, please could you let me know if you're happy with the editing changes.
Thanks.

**Comment by ksloven** [comment]
I'm happy with the editing changes. Thank you very much

**Comment by anselld** [comment]
@ksloven 
@hansieodendaal 
It's a pleasure - thanks for your help, Kenny.

**Comment by anselld** [comment]
@hansieodendaal 
Fixed 120 column width. Please let me know if the file is okay. Thanks.

**Comment by hansieodendaal** [comment]
Hi @anselld, fixed 2 links and showed where the line widths could still be applied. See latest commits.

**Pull request merged**